umh-core/pkg/generated/status-message.go

```diff
@@ -0,0 +1,317 @@
+// To parse and unparse this JSON data, add this code to your project and do:
+//
+//    statusMessage, err := UnmarshalStatusMessage(bytes)
+//    bytes, err = statusMessage.Marshal()
+
+package generated
+
+import "bytes"
+import "errors"
+
+import "encoding/json"
+
+func UnmarshalStatusMessage(data []byte) (StatusMessage, error) {
+	var r StatusMessage
+	err := json.Unmarshal(data, &r)
+	return r, err
+}
+
+func (r *StatusMessage) Marshal() ([]byte, error) {
+	return json.Marshal(r)
+}
+
+// Schema for the status message containing system state and DFC information
+type StatusMessage struct {
+	Core    Core                   `json:"core"`
+	General General                `json:"general"`
+	Plugins map[string]interface{} `json:"plugins"`
+}
+
+type Core struct {
+	Agent             Agent                  `json:"agent"`
+	Container         Container              `json:"container"`
+	Dfcs              []Dfc                  `json:"dfcs"`
+	EventsTable       map[string]EventsTable `json:"eventsTable,omitempty"`
+	Latency           Latency                `json:"latency"`
+	Redpanda          Redpanda               `json:"redpanda"`
+	ReleaseChannel    string                 `json:"releaseChannel"`
+	SupportedFeatures []string               `json:"supportedFeatures"`
+	UnsTable          map[string]UnsTable    `json:"unsTable,omitempty"`
+	Version           string                 `json:"version"`
+}
+
+type Agent struct {
+	Health Health `json:"health"`
+}
+
+type Health struct {
+	Message string `json:"message"`
+	State   string `json:"state"`
+}
+
+type Container struct {
+	Architecture string `json:"architecture"`
+	CPU          CPU    `json:"cpu"`
+	Disk         Disk   `json:"disk"`
+	Hwid         string `json:"hwid"`
+	Memory       Memory `json:"memory"`
+}
+
+type CPU struct {
+	Health Health  `json:"health"`
+	Limit  float64 `json:"limit"`
+	Usage  float64 `json:"usage"`
+}
+
+type Disk struct {
+	Health Health  `json:"health"`
+	Limit  float64 `json:"limit"`
+	Usage  float64 `json:"usage"`
+}
+
+type Memory struct {
+	Health Health  `json:"health"`
+	Limit  float64 `json:"limit"`
+	Usage  float64 `json:"usage"`
+}
+
+type Dfc struct {
+	Connections        []Connection `json:"connections,omitempty"`
+	CurrentVersionUUID *string      `json:"currentVersionUUID"`
+	DeploySuccess      bool         `json:"deploySuccess"`
+	Health             *Health      `json:"health"`
+	Metrics            *DFCMetrics  `json:"metrics"`
+	Name               *string      `json:"name"`
+	Type               Type         `json:"type"`
+	UUID               string       `json:"uuid"`
+	DataContract       *string      `json:"dataContract,omitempty"`
+	InputType          *string      `json:"inputType,omitempty"`
+	IsReadOnly         *bool        `json:"isReadOnly,omitempty"`
+	OutputType         *string      `json:"outputType,omitempty"`
+}
+
+type Connection struct {
+	Health  Health  `json:"health"`
+	Host    string  `json:"host"`
+	Latency float64 `json:"latency"`
```

Comment:
what does this field mean? we also have a struct with latency. what type of latency is it?